src/Main.elm

```diff
@@ -1,10 +1,199 @@
+import Dict                     exposing (Dict)
+import Html                     exposing (Html, div, text)
+import Html.Attributes          exposing (style)
+import Html.Events              exposing (onClick)
+import Http
+import Json.Decode as Decode    exposing (Decoder, decodeString)
+import Task                     exposing (Task)
+
+
+type alias Model =
+    { audiences: Dict Int (List Audience)
+    , folders: Dict Int (List AudienceFolder)
+    , stack: List AudienceFolder
+    }
+
+
+type Msg
+    = GotAudiences (List Audience)
+    | GotAudienceFolders (List AudienceFolder)
+    | OpenFolder AudienceFolder
+    | GoUp
+
+
+-- MAIN
 
-main : Html msg
 main =
-    Html.text "There will be app soon!"
+    Browser.element
+        { init          = init
+        , view          = view
+        , update        = update
+        , subscriptions = always Sub.none
+        }
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( { audiences = Dict.empty
+      , folders = Dict.empty
+      , stack = []
+      }
+
+    -- Load audiences and folders concurrently.
+    --
+    -- The real application will execute an appropiate HTTP requests.
+    -- But for testing purposes we will use the provided fixtures.
+    , Cmd.batch
+        [ Task.attempt (Result.withDefault [] >> GotAudiences)
+              <| loadDataTask decodeGetAudiences audiencesJSON
+        , Task.attempt (Result.withDefault [] >> GotAudienceFolders)
+              <| loadDataTask decodeGetAudienceFolders audienceFoldersJSON
+        ]
+    )
+
+
+loadDataTask : Decoder a -> String -> Task Http.Error a
+loadDataTask decoder fixture =
+        case decodeString decoder fixture of
+            Ok data ->
+                Task.succeed data
+
+            -- map JSON decode error into Http.BadBody as it done by elm/http
+            Err err ->
+                Task.fail <| Http.BadBody <| Decode.errorToString err
+
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        GotAudiences audiences ->
+            ( { model
+                  | stack = []
+                  , audiences = buildAudiences audiences
+              }
+            , Cmd.none
+            )
+
+        GotAudienceFolders folders ->
+            ( { model
+                  | stack = []
+                  , folders = buildFolders folders
+              }
+            , Cmd.none
+            )
+
+        OpenFolder folder ->
+            ( { model
+                  | stack = folder :: model.stack
+              }
+            , Cmd.none
+            )
+
+        GoUp ->
+            ( { model | stack = List.tail model.stack |> Maybe.withDefault [] }
+            , Cmd.none
+            )
+
+
+view : Model -> Html Msg
+view model =
+    let parent =
+            List.head model.stack
+
+        parentId =
+            case parent of
+                Nothing ->
+                    0
+                Just folder ->
+                    folder.id
```

Comment:
This could be potentially danger approach, since `0` is valid integer and what if you will have such a number on BE? In this case we are safe since we own BE service (or BE teams does). In comments earlier you have probably better example with "null" string for `Nothing`.
```
case parentId if
     Nothing ->
         "null"
     Just id ->
         String.fromInt id
```

Reply:
I agree that this is dangerous and there is a  comment in src/Tree.elm:6 with a remainder to myself with similar code.